presto-main/src/main/java/io/prestosql/server/protocol/RowIterable.java

```diff
@@ -74,9 +82,34 @@ private RowIterator(ConnectorSession session, List<Type> types, Page page)
             for (int channel = 0; channel < page.getChannelCount(); channel++) {
                 Type type = types.get(channel);
                 Block block = page.getBlock(channel);
-                values.add(type.getObjectValue(session, block, position));
+
+                Object value = type.getObjectValue(session, block, position);
+
+                if (value != null && type instanceof TimestampType) {
+                    value = renderTimestamp((TimestampType) type, (SqlTimestamp) value);
+                }
+
+                values.add(value);
             }
             return Collections.unmodifiableList(values);
         }
+
+        private Object renderTimestamp(TimestampType type, SqlTimestamp timestamp)
+        {
+            ZoneId zoneId;
+            if (session.isLegacyTimestamp()) {
+                zoneId = ZoneId.of(session.getTimeZoneKey().getId());
```

Comment:
We should add ZoneID to DateTimeZoneIndex

Reply:
I already added it, but forgot to update this callsite